openshift/message-board/message-service/src/main/java/org/hibernate/demo/message/post/core/service/procedure/RemoteProcedureRegister.java

```diff
@@ -0,0 +1,26 @@
+package org.hibernate.demo.message.post.core.service.procedure;
+
+import javax.ejb.Singleton;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.StoredProcedureQuery;
+
+@Singleton
+public class RemoteProcedureRegister {
+
+	@Inject
+	private EntityManager em;
+
+	protected boolean init = false;
+
+	// a javax.ejb.Singleton should be already synchronized
+	public void registerMarshaller() {
+		if ( init ) {
+			return;
+		}
+
+		StoredProcedureQuery storedProcedureQuery = em.createStoredProcedureQuery( "RegisterMarshallersTask" );
+		storedProcedureQuery.execute();
+		init = true;
```

Comment:
if the task is called `Register-` this boolean becomes clearer if it's called `registered`. Also the name of the method is a bit confusing because the class is called `*Procedure*` but it actually deals with *Marshallers*

Reply:
With Infinispan we use a task to register the marshallers we need to apply on server.